### ISSUE TITLE
Error initializing AttributeResolver

### ISSUE THREAD

**Issue by iromli** [GluuFederation/docker-oxshibboleth#6]
Getting startup error when running oxShibboleth.

From `idp-warn.log`:

```
2019-07-23 17:10:31,858 -  - WARN [net.shibboleth.utilities.java.support.net.CookieManager:171] - Use of secure property is strongly advised
2019-07-23 17:10:32,041 -  - WARN [net.shibboleth.utilities.java.support.net.CookieManager:171] - Use of secure property is strongly advised
2019-07-23 17:10:33,324 -  - WARN [DEPRECATED:118] - XML Element 'SourceAttribute', (file [/opt/shibboleth-idp/conf/attribute-resolver.xml]): This wil
l be removed in the next major version of this software; replacement is by using <InputAttributeDefinition> and <InputDataConnector>
2019-07-23 17:10:36,835 -  - WARN [net.shibboleth.utilities.java.support.net.CookieManager:171] - Use of secure property is strongly advised
2019-07-23 17:24:57,787 -  - WARN [net.shibboleth.utilities.java.support.net.CookieManager:171] - Use of secure property is strongly advised
2019-07-23 17:24:57,827 -  - WARN [net.shibboleth.utilities.java.support.net.CookieManager:171] - Use of secure property is strongly advised
2019-07-23 17:24:58,185 -  - WARN [DEPRECATED:122] - XML Element 'dc:StartTLSTrustCredential', (file [/opt/shibboleth-idp/conf/attribute-resolver.xml]
): This will be removed in the next major version of this software
2019-07-23 17:24:58,203 -  - WARN [DEPRECATED:118] - xsi:type '{urn:mace:shibboleth:2.0:resolver:dc}LDAPDirectory', (file [/opt/shibboleth-idp/conf/at
tribute-resolver.xml]): This will be removed in the next major version of this software; replacement is {urn:mace:shibboleth:2.0:resolver}LDAPDirector
y
2019-07-23 17:24:58,912 -  - ERROR [net.shibboleth.idp.attribute.resolver.dc.ldap.impl.ConnectionFactoryValidator:156] - Connection factory validation failed
org.ldaptive.OperationException: javax.naming.CommunicationException: ldap:1636 [Root exception is java.lang.NullPointerException: Thread local SslConfig has not been set]
        at org.ldaptive.provider.ProviderUtils.throwOperationException(ProviderUtils.java:67)
```

Environment:

```
Runtime: Docker
OS: Alpine
JVM: AdoptOpenJDK 11.0.4_11
```

**Comment by iromli** [comment]
Here's the full logs of `/opt/shibboleth-idp/logs/idp-warn.log`:

```
2019-07-23 17:10:31,858 -  - WARN [net.shibboleth.utilities.java.support.net.CookieManager:171] - Use of secure property is strongly advised
2019-07-23 17:10:32,041 -  - WARN [net.shibboleth.utilities.java.support.net.CookieManager:171] - Use of secure property is strongly advised
2019-07-23 17:10:33,324 -  - WARN [DEPRECATED:118] - XML Element 'SourceAttribute', (file [/opt/shibboleth-idp/conf/attribute-resolver.xml]): This will be removed in the next major version of this software; replacement is by using <InputAttributeDefinition> and <InputDataConnector>
2019-07-23 17:10:36,835 -  - WARN [net.shibboleth.utilities.java.support.net.CookieManager:171] - Use of secure property is strongly advised
2019-07-23 17:24:57,787 -  - WARN [net.shibboleth.utilities.java.support.net.CookieManager:171] - Use of secure property is strongly advised
2019-07-23 17:24:57,827 -  - WARN [net.shibboleth.utilities.java.support.net.CookieManager:171] - Use of secure property is strongly advised
2019-07-23 17:24:58,185 -  - WARN [DEPRECATED:122] - XML Element 'dc:StartTLSTrustCredential', (file [/opt/shibboleth-idp/conf/attribute-resolver.xml]): This will be removed in the next major version of this software
2019-07-23 17:24:58,203 -  - WARN [DEPRECATED:118] - xsi:type '{urn:mace:shibboleth:2.0:resolver:dc}LDAPDirectory', (file [/opt/shibboleth-idp/conf/attribute-resolver.xml]): This will be removed in the next major version of this software; replacement is {urn:mace:shibboleth:2.0:resolver}LDAPDirectory
2019-07-23 17:24:58,912 -  - ERROR [net.shibboleth.idp.attribute.resolver.dc.ldap.impl.ConnectionFactoryValidator:156] - Connection factory validation failed
org.ldaptive.OperationException: javax.naming.CommunicationException: ldap:1636 [Root exception is java.lang.NullPointerException: Thread local SslConfig has not been set]
	at org.ldaptive.provider.ProviderUtils.throwOperationException(ProviderUtils.java:67)
Caused by: javax.naming.CommunicationException: ldap:1636
	at java.naming/com.sun.jndi.ldap.Connection.<init>(Unknown Source)
Caused by: java.lang.NullPointerException: Thread local SslConfig has not been set
	at org.ldaptive.ssl.ThreadLocalTLSSocketFactory.getDefault(ThreadLocalTLSSocketFactory.java:70)
2019-07-23 17:24:58,920 -  - ERROR [net.shibboleth.idp.attribute.resolver.dc.ldap.impl.LDAPDataConnector:151] - Data Connector 'siteDataConnector': Invalid connector configuration
net.shibboleth.idp.attribute.resolver.dc.ValidationException: [org.ldaptive.OperationException@1974062116::resultCode=PROTOCOL_ERROR, matchedDn=null, responseControls=null, referralURLs=null, messageId=-1, message=javax.naming.CommunicationException: ldap:1636 [Root exception is java.lang.NullPointerException: Thread local SslConfig has not been set], providerException=javax.naming.CommunicationException: ldap:1636 [Root exception is java.lang.NullPointerException: Thread local SslConfig has not been set]]
	at net.shibboleth.idp.attribute.resolver.dc.ldap.impl.ConnectionFactoryValidator.validate(ConnectionFactoryValidator.java:158)
Caused by: org.ldaptive.OperationException: javax.naming.CommunicationException: ldap:1636 [Root exception is java.lang.NullPointerException: Thread local SslConfig has not been set]
	at org.ldaptive.provider.ProviderUtils.throwOperationException(ProviderUtils.java:67)
Caused by: javax.naming.CommunicationException: ldap:1636
	at java.naming/com.sun.jndi.ldap.Connection.<init>(Unknown Source)
Caused by: java.lang.NullPointerException: Thread local SslConfig has not been set
	at org.ldaptive.ssl.ThreadLocalTLSSocketFactory.getDefault(ThreadLocalTLSSocketFactory.java:70)
2019-07-23 17:24:58,921 -  - WARN [net.shibboleth.ext.spring.context.FilesystemGenericApplicationContext:551] - Exception encountered during context initialization - cancelling refresh attempt: org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'siteDataConnector': Invocation of init method failed; nested exception is net.shibboleth.utilities.java.support.component.ComponentInitializationException: Data Connector 'siteDataConnector': Invalid connector configuration
2019-07-23 17:24:58,922 -  - ERROR [net.shibboleth.utilities.java.support.service.AbstractReloadableService:182] - Service 'shibboleth.AttributeResolverService': Initial load failed
net.shibboleth.utilities.java.support.service.ServiceException: org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'siteDataConnector': Invocation of init method failed; nested exception is net.shibboleth.utilities.java.support.component.ComponentInitializationException: Data Connector 'siteDataConnector': Invalid connector configuration
	at net.shibboleth.ext.spring.service.ReloadableSpringService.doReload(ReloadableSpringService.java:377)
Caused by: org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'siteDataConnector': Invocation of init method failed; nested exception is net.shibboleth.utilities.java.support.component.ComponentInitializationException: Data Connector 'siteDataConnector': Invalid connector configuration
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.initializeBean(AbstractAutowireCapableBeanFactory.java:1631)
Caused by: net.shibboleth.utilities.java.support.component.ComponentInitializationException: Data Connector 'siteDataConnector': Invalid connector configuration
	at net.shibboleth.idp.attribute.resolver.dc.ldap.impl.LDAPDataConnector.doInitialize(LDAPDataConnector.java:152)
Caused by: net.shibboleth.idp.attribute.resolver.dc.ValidationException: [org.ldaptive.OperationException@1974062116::resultCode=PROTOCOL_ERROR, matchedDn=null, responseControls=null, referralURLs=null, messageId=-1, message=javax.naming.CommunicationException: ldap:1636 [Root exception is java.lang.NullPointerException: Thread local SslConfig has not been set], providerException=javax.naming.CommunicationException: ldap:1636 [Root exception is java.lang.NullPointerException: Thread local SslConfig has not been set]]
	at net.shibboleth.idp.attribute.resolver.dc.ldap.impl.ConnectionFactoryValidator.validate(ConnectionFactoryValidator.java:158)
Caused by: org.ldaptive.OperationException: javax.naming.CommunicationException: ldap:1636 [Root exception is java.lang.NullPointerException: Thread local SslConfig has not been set]
	at org.ldaptive.provider.ProviderUtils.throwOperationException(ProviderUtils.java:67)
Caused by: javax.naming.CommunicationException: ldap:1636
	at java.naming/com.sun.jndi.ldap.Connection.<init>(Unknown Source)
Caused by: java.lang.NullPointerException: Thread local SslConfig has not been set
	at org.ldaptive.ssl.ThreadLocalTLSSocketFactory.getDefault(ThreadLocalTLSSocketFactory.java:70)
2019-07-23 17:25:00,403 -  - WARN [net.shibboleth.ext.spring.context.DelimiterAwareApplicationContext:551] - Exception encountered during context initialization - cancelling refresh attempt: org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'shibboleth.metrics.RegisterMetricSets$child#0' defined in URL [file:/opt/shibboleth-idp/conf/admin/metrics.xml]: Cannot resolve reference to bean 'shibboleth.metrics.AttributeResolverGaugeSet' while setting bean property 'arguments' with key [7]; nested exception is org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'shibboleth.metrics.AttributeResolverGaugeSet' defined in URL [file:/opt/shibboleth-idp/system/conf/general-admin-system.xml]: Invocation of init method failed; nested exception is net.shibboleth.utilities.java.support.component.ComponentInitializationException: Injected service was null or not an AttributeResolver
2019-07-23 17:25:00,464 -  - ERROR [org.springframework.web.context.ContextLoader:350] - Context initialization failed
org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'shibboleth.metrics.RegisterMetricSets$child#0' defined in URL [file:/opt/shibboleth-idp/conf/admin/metrics.xml]: Cannot resolve reference to bean 'shibboleth.metrics.AttributeResolverGaugeSet' while setting bean property 'arguments' with key [7]; nested exception is org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'shibboleth.metrics.AttributeResolverGaugeSet' defined in URL [file:/opt/shibboleth-idp/system/conf/general-admin-system.xml]: Invocation of init method failed; nested exception is net.shibboleth.utilities.java.support.component.ComponentInitializationException: Injected service was null or not an AttributeResolver at org.springframework.beans.factory.support.BeanDefinitionValueResolver.resolveReference(BeanDefinitionValueResolver.java:359)
Caused by: org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'shibboleth.metrics.AttributeResolverGaugeSet' defined in URL [file:/opt/shibboleth-idp/system/conf/general-admin-system.xml]: Invocation of init method failed; nested exception is net.shibboleth.utilities.java.support.component.ComponentInitializationException: Injected service was null or not an AttributeResolver at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.initializeBean(AbstractAutowireCapableBeanFactory.java:1631)
Caused by: net.shibboleth.utilities.java.support.component.ComponentInitializationException: Injected service was null or not an AttributeResolver at
net.shibboleth.idp.attribute.resolver.impl.AttributeResolverServiceGaugeSet.doInitialize(AttributeResolverServiceGaugeSet.java:104)
```

**Comment by iromli** [comment]
Fixed at 92f39eb29d8dc7a424146e7cf1a10b8df5a39b23.
Reference: https://wiki.shibboleth.net/confluence/display/IDP30/LDAPonJava%3E8
